src/main/g8/src/main/scala/$package$/Boot.scala

```diff
@@ -1,87 +1,65 @@
+import com.typesafe.config.{ Config, ConfigFactory }
+import zio.clock.Clock
 import zio.config.typesafe.TypesafeConfig
-import zio.config.{ config, Config }
 import zio.console._
-import zio.{ App, Has, ZIO, ZLayer, ZManaged }
 import zio.logging._
 import zio.logging.slf4j._
-import zio.{ App, Has, TaskLayer, ULayer, ZIO, ZLayer, ZManaged }
-import zio.clock.Clock
+import zio._
 
 object Boot extends App {
 
-  val program: ZIO[HttpServer with Console, Throwable, Unit] =
-    HttpServer.start.use { binding =>
-      for {
-        _ <- putStrLn(
-              s"Server online. Press RETURN to stop..."
-            )
-        _ <- getStrLn
-      } yield ()
-    }
-
-  def bindAndHandle(routes: Route, host: String, port: Int)(
-    implicit system: ActorSystem
-  ): ZManaged[Any, Throwable, Http.ServerBinding] =
-    ZManaged.make(ZIO.fromFuture(_ => Http().bindAndHandle(routes, host, port)))(b =>
-      ZIO.fromFuture(_ => b.unbind()).orDie
-    )
+  def run(args: List[String]): ZIO[zio.ZEnv, Nothing, Int] =
+    ZIO(ConfigFactory.load.resolve)
+      .flatMap(rawConfig => program.provideCustomLayer(prepareEnvironment(rawConfig)))
+      .as(0)
+      .catchAll(error => putStrLn(error.getMessage).as(1))
 
-  val loadConfig = ZIO.effect(ConfigFactory.load.resolve)
+  private val program: ZIO[HttpServer with Console, Throwable, Unit] =
+    HttpServer.start.use(_ => putStrLn(s"Server online. Press RETURN to stop...") <* getStrLn)
 
-  val actorSystem = ZLayer.fromManaged(
-    ZManaged.make(ZIO.effect(ActorSystem("$name$-system")))(s => ZIO.fromFuture(_ => s.terminate()).either)
-  )
+  private def prepareEnvironment(rawConfig: Config): TaskLayer[HttpServer] = {
+    val configLayer = TypesafeConfig.fromTypesafeConfig(rawConfig, AppConfig.descriptor)
 
-  val loggingLayer: ULayer[Logging] = Slf4jLogger.make { (context, message) =>
-    val logFormat = "[correlation-id = %s] %s"
-    val correlationId = LogAnnotation.CorrelationId.render(
-      context.get(LogAnnotation.CorrelationId)
-    )
-    logFormat.format(correlationId, message)
-  }
+    // using raw config since it's recommended and the simplest to work with slick
+    val dbConfigLayer = ZLayer.fromEffect(ZIO(rawConfig.getConfig("db")))
 
-  val actorSystemLayer: TaskLayer[Has[ActorSystem]] = ZLayer.fromManaged(
-    ZManaged.make(ZIO.effect(ActorSystem("zio-example-system")))(s => ZIO.fromFuture(_ => s.terminate()).either)
-  )
+    // narrowing down to the required part of the config to ensure separation of concerns
+    val apiConfigLayer = configLayer.map(c => Has(c.get.api))
 
-  def run(args: List[String]): ZIO[zio.ZEnv, Nothing, Int] =
-    loadConfig.flatMap { rawConfig =>
-      val configLayer = TypesafeConfig.fromTypesafeConfig(rawConfig, AppConfig.descriptor)
+    val actorSystemLayer: TaskLayer[Has[ActorSystem]] = ZLayer.fromManaged {
+      ZManaged.make(ZIO(ActorSystem("zio-akka-quickstart-system")))(s => ZIO.fromFuture(_ => s.terminate()).either)
```

Comment:
Why not still use the template variable for system name?

Reply:
Yeah, you're right, I've just missed that